src/components/views/rooms/wysiwyg_composer/hooks/useSuggestion.ts

```diff
@@ -17,25 +17,37 @@ limitations under the License.
 import { Attributes, MappedSuggestion, SuggestionChar, SuggestionType } from "@matrix-org/matrix-wysiwyg";
 import { SyntheticEvent, useState } from "react";
 
-// This type is a close approximation of what we use in the Rust model for the rich version of the
-// editor, we use this to try and make this simple model as similar as possible to allow reuse of
-// the autocomplete component
+/* To allow reuse of the autocomplete component, we need to record similar information as used in 
+the Rust model (keyChar, type and text) as well as information required to allow us to do the 
+DOM manipulation when replacing the suggestion with a mention or command (node, start and
+end offsets). */
 export type PlainTextSuggestionPattern = {
```

Comment:
I'm sorry to bang on about this, but this still doesn't quite join the dots. I understand that we need to record the information you mention, but how does this type help with that process?

Is this correct?

```suggestion
/**
  * Information about the current state of the `useSuggestion` hook.
  */
export type PlainTextSuggestionPattern = {
    /* 
     * To allow reuse of the autocomplete component, we need to record similar information as used in 
     * the Rust model (keyChar, type and text) as well as information required to allow us to do the 
     * DOM manipulation when replacing the suggestion with a mention or command (node, start and
     * end offsets).
     */
```

(If so, why is it called `PlainTextSuggestionPattern` rather than, say, `SuggestionState` ?)

Reply:
I agree that `SuggestionState` is a better name for it. It was named the way it was as I was coming over from our Rust naming and keeping it similar, but really there's no need for it to be like that.

I've changed the name of the type to be `Suggestion` and then `SuggestionState` is `Suggestion | null` which I think makes sense. I've also tidied up the type and changed the comment to reflect this update. Let me know if you think this is an improvement.

Reply:
much clearer, thank you!